### PR TITLE
DEV: adds a new dropdown widget usable in any widget

### PR DESCRIPTION
**Pull request by jjaffeux**
image

Note this commit needs: https://github.com/discourse/discourse/pull/9296 tests will fail until this is merged.

### CODE REVIEW

File: app/assets/javascripts/discourse/widgets/dropdown.js [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+
+createWidget("dropdown-header", DropdownHeaderClass);
+
+export const DropdownItemClass = {
+  tagName: "div",
+
+  transform(attrs) {
+    return {
+      content:
+        attrs.item === "separator"
```
Comment (eviltrout) [inline app/assets/javascripts/discourse/widgets/dropdown.js]: I can't decide if I hate this or not :smiley: 

On one hand nested ternary operators make my brain hurt, but on the other I fully understand what this is doing. I am leaning towards "let's keep it."

File: app/assets/javascripts/discourse/widgets/dropdown.js [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+      [
+        { id: 1, label: "foo.bar" },
+        "separator",
+        { id: 2, translatedLabel: "FooBar" },
+        { id: 3 label: "foo.baz", icon: "times" }
+      ]
+      ```
+
+    - options: accepts a hash of optional attributes
+      - smallHeader: true/false, if true "btn-small" class will be added to header.
```
Comment (CvX) [inline app/assets/javascripts/discourse/widgets/dropdown.js]: A bool for adding a class doesn't scale well if you'd ever want to add another class to that header/button. Wouldn't explicitly passing a class list be better?

File: app/assets/javascripts/discourse/widgets/dropdown.js [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+    )
+  }}
+
+  Mandatory attributes:
+
+    - id: must be unique in the application
+
+    - label: an i18n key to be translated and displayed on the header
+
+    - translatedLabel: an already translated label to display on the header
```
[the remark this answers is not in the document]
Comment (jjaffeux) [reply app/assets/javascripts/discourse/widgets/dropdown.js]: Yes right will clarify this :+1:

File: app/assets/javascripts/discourse/widgets/dropdown.js (hunk 1 above)
Comment (jjaffeux) [reply app/assets/javascripts/discourse/widgets/dropdown.js]: Yes I feel the same :D

File: app/assets/javascripts/discourse/widgets/dropdown.js (hunk 2 above)
Comment (jjaffeux) [reply app/assets/javascripts/discourse/widgets/dropdown.js]: yes agreed it feels a little bit hacky... I just thought this is a very common case, so I could make it explicit as `btn-small` requires you to know the internals of the dropdown

**Pull request merged**